src/code/z_vimode.c

```diff
@@ -1,17 +1,214 @@
+    _LOG(89, "osvimodep->comRegs.width = %08x\n", (*viMode_ptr)->comRegs.width);
+    _LOG(90, "osvimodep->comRegs.burst = %08x\n", (*viMode_ptr)->comRegs.burst);
+    _LOG(91, "osvimodep->comRegs.vSync = %08x\n", (*viMode_ptr)->comRegs.vSync);
+    _LOG(92, "osvimodep->comRegs.hSync = %08x\n", (*viMode_ptr)->comRegs.hSync);
+    _LOG(93, "osvimodep->comRegs.leap = %08x\n", (*viMode_ptr)->comRegs.leap);
+    _LOG(94, "osvimodep->comRegs.hStart = %08x\n", (*viMode_ptr)->comRegs.hStart);
+    _LOG(95, "osvimodep->comRegs.xScale = %08x\n", (*viMode_ptr)->comRegs.xScale);
+    _LOG(96, "osvimodep->fldRegs[0].vStart = %08x\n", (*viMode_ptr)->fldRegs[0].vStart);
+    _LOG(97, "osvimodep->fldRegs[0].vBurst = %08x\n", (*viMode_ptr)->fldRegs[0].vBurst);
+    _LOG(98, "osvimodep->fldRegs[0].origin = %08x\n", (*viMode_ptr)->fldRegs[0].origin);
+    _LOG(99, "osvimodep->fldRegs[0].yScale = %08x\n", (*viMode_ptr)->fldRegs[0].yScale);
+    _LOG(100, "osvimodep->fldRegs[0].vIntr = %08x\n", (*viMode_ptr)->fldRegs[0].vIntr);
+    _LOG(101, "osvimodep->fldRegs[1].vStart = %08x\n", (*viMode_ptr)->fldRegs[1].vStart);
+    _LOG(102, "osvimodep->fldRegs[1].vBurst = %08x\n", (*viMode_ptr)->fldRegs[1].vBurst);
+    _LOG(103, "osvimodep->fldRegs[1].origin = %08x\n", (*viMode_ptr)->fldRegs[1].origin);
+    _LOG(104, "osvimodep->fldRegs[1].yScale = %08x\n", (*viMode_ptr)->fldRegs[1].yScale);
+    _LOG(105, "osvimodep->fldRegs[1].vIntr = %08x\n", (*viMode_ptr)->fldRegs[1].vIntr);
+#undef _LOG
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC2F4.s")
+// This function configures the custom VI mode (`viMode.customViMode`) based
+// on the other flags in `viMode`.
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/ViMode_Configure.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC89C.s")
+void ViMode_Save(ViMode* viMode) {
+    SREG(48) = viMode->viModeBase;
+    SREG(49) = viMode->viWidth;
+    SREG(50) = viMode->viHeight;
+    SREG(51) = viMode->unk64;
+    SREG(52) = viMode->unk60;
+    SREG(53) = viMode->unk5C;
+    SREG(54) = viMode->unk58;
+    if (SREG(58) == 1) {
+        SREG(58) = 0;
+        switch (SREG(59)) {
+            case 1:
+                osSyncPrintf("osViModePalLan1\n");
+                ViMode_LogPrint(&osViModePalLan1);
+                break;
+            case 2:
+                osSyncPrintf("osViModeFpalLan1\n");
+                ViMode_LogPrint(&osViModeFpalLan1);
+                break;
+            default:
+                osSyncPrintf("Custom\n");
+                ViMode_LogPrint(&viMode->customViMode);
+                break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC9A4.s")
+void ViMode_Load(ViMode* viMode) {
+    GameInfo* _gameInfo = gGameInfo;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800ACA28.s")
+    // NB: This condition is always false
+    s16 width = SREG(49);
+    if ((width & ((s16)0xfffc)) == 1) {
```

Comment:
```suggestion
    s16 width = SREG(49);
    // NB: This condition is always false
    if ((width & ((s16)0xfffc)) == 1) {
```

Reply:
Also, I don't think this is the correct way to flag this, but I also don't know what our standard is.